### PR TITLE
May Security Update

### PR DESCRIPTION
**Pull request by willronchetti**
- Branch to branch PR from auth0 to technical-review (will isolate change if necessary)
- Allows Python 3.8
- Updates Docker image to 3.8
- Add supervisord support
- Add Auth0 options to settings

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -114,24 +113,25 @@ RUN chown nginx:nginx development.ini
...
-RUN touch production.ini
-RUN touch session-secret.b64
-RUN chown nginx:nginx session-secret.b64
-RUN chown nginx:nginx production.ini
-RUN chown nginx:nginx poetry.toml
+RUN touch production.ini && chown nginx:nginx production.ini && \
+    touch session-secret.b64 && chown nginx:nginx session-secret.b64 && chown nginx:nginx poetry.toml && \
+    touch supervisord.log && chown nginx:nginx supervisord.log && \
+    touch supervisord.sock && chown nginx:nginx supervisord.sock && \
+    touch supervisord.pid && chown nginx:nginx supervisord.pid
```
Comment (netsettler) [inline Dockerfile:120]: Do the poetry.toml setting at the start or end, as it breaks up the visual rhythm being in the middle.
Honestly, I would probably put the chowns on their own line, even at the vertical cost of more space (or align them so they all start at the same place on line)? Please double-check the following (untested) for typos:
```suggestion
RUN chown nginx:nginx poetry.toml && \
    touch production.ini && \
    chown nginx:nginx production.ini && \
    touch session-secret.b64 && \
    chown nginx:nginx session-secret.b64 &&  \
    touch supervisord.log && \
    chown nginx:nginx supervisord.log && \
    touch supervisord.sock && \
    chown nginx:nginx supervisord.sock && \
    touch supervisord.pid && \
    chown nginx:nginx supervisord.pid
```

File: deploy/docker/production/supervisord.conf [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+redirect_stderr=true
+
+[program:cgap5]
+autorestart=true
+startsecs=6
+command=pserve production.ini http_port=6547
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile_maxbytes=0
+redirect_stderr=true
```
Comment (netsettler) [inline deploy/docker/production/supervisord.conf:54]: Needs a final newline.
```suggestion
redirect_stderr=true

```

**Review by netsettler — APPROVED** [review]
My comments are not blocking.

**Comment by willronchetti** [comment]
Closing in favor of #610

**Pull request closed without merging**